src/Gaufrette/Adapter/AzureBlobStorage.php

```diff
@@ -403,4 +408,53 @@ private function guessContentType($content)
 
         return $fileInfo->buffer($content);
     }
+
+    /**
+     * Error message to be parsed.
+     *
+     * @param  ResponseInterface $response The response with a response body.
+     *
+     * @return string
+     */
+    protected static function parseErrorCode(ResponseInterface $response)
+    {
+        //try to parse using xml serializer, if failed, return the whole body
+        //as the error message.
+        try {
+            $sxml = new \SimpleXMLElement($response->getBody());
+            $data = static::_sxml2arr($sxml);
```

Comment:
Since we are only interested in the very first level of `$sxml`, I guess we don't need to recursively convert the object.